integration-tests/src/test/kotlin/org/stellar/anchor/platform/test/Sep31CustodyRpcEnd2EndTests.kt

```diff
@@ -0,0 +1,471 @@
+package org.stellar.anchor.platform.test
+
+import com.google.gson.reflect.TypeToken
+import io.ktor.client.plugins.*
+import io.ktor.http.*
+import junit.framework.TestCase.assertEquals
+import junit.framework.TestCase.fail
+import kotlin.time.Duration.Companion.seconds
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.runBlocking
+import org.junit.jupiter.api.Assertions.assertNotNull
+import org.skyscreamer.jsonassert.JSONAssert
+import org.stellar.anchor.api.event.AnchorEvent
+import org.stellar.anchor.api.sep.SepTransactionStatus
+import org.stellar.anchor.api.sep.sep12.Sep12PutCustomerRequest
+import org.stellar.anchor.api.sep.sep31.Sep31GetTransactionResponse
+import org.stellar.anchor.api.sep.sep31.Sep31PostTransactionRequest
+import org.stellar.anchor.apiclient.PlatformApiClient
+import org.stellar.anchor.auth.AuthHelper
+import org.stellar.anchor.platform.*
+import org.stellar.anchor.util.GsonUtils
+import org.stellar.anchor.util.Log.info
+import org.stellar.anchor.util.MemoHelper
+import org.stellar.anchor.util.Sep1Helper
+import org.stellar.anchor.util.StringHelper.json
+import org.stellar.reference.client.AnchorReferenceServerClient
+import org.stellar.reference.wallet.WalletServerClient
+import org.stellar.walletsdk.ApplicationConfiguration
+import org.stellar.walletsdk.StellarConfiguration
+import org.stellar.walletsdk.Wallet
+import org.stellar.walletsdk.anchor.MemoType
+import org.stellar.walletsdk.asset.IssuedAssetId
+import org.stellar.walletsdk.asset.StellarAssetId
+import org.stellar.walletsdk.horizon.SigningKeyPair
+import org.stellar.walletsdk.horizon.sign
+
+class Sep31CustodyRpcEnd2EndTests(
+  config: TestConfig,
+  val toml: Sep1Helper.TomlContent,
+  val jwt: String
+) {
+  private val gson = GsonUtils.getInstance()
+  private val walletSecretKey = System.getenv("WALLET_SECRET_KEY") ?: CLIENT_WALLET_SECRET
+  private val keypair = SigningKeyPair.fromSecret(walletSecretKey)
+  private val wallet =
+    Wallet(
+      StellarConfiguration.Testnet,
+      ApplicationConfiguration { defaultRequest { url { protocol = URLProtocol.HTTP } } }
+    )
+  private val maxTries = 90
+  private val anchorReferenceServerClient =
+    AnchorReferenceServerClient(Url(config.env["reference.server.url"]!!))
+  private val walletServerClient = WalletServerClient(Url(config.env["wallet.server.url"]!!))
+  private val sep12Client = Sep12Client(toml.getString("KYC_SERVER"), jwt)
+  private val sep31Client = Sep31Client(toml.getString("DIRECT_PAYMENT_SERVER"), jwt)
+  private val sep38Client = Sep38Client(toml.getString("ANCHOR_QUOTE_SERVER"), jwt)
+  private val platformApiClient =
+    PlatformApiClient(AuthHelper.forNone(), config.env["platform.server.url"]!!)
+
+  private fun compareAndAssertEvents(
+    asset: StellarAssetId,
+    expectedEvents: List<AnchorEvent>,
+    actualEvents: List<AnchorEvent>
+  ) {
+    expectedEvents.forEachIndexed { index, expectedEvent ->
```

Comment:
Why do we modify expectedEvent before comparing it to actualEvent?

Reply:
Because some fields are dynamic(for example dates) and can not be specified in `expectedEvent`. The same logic is used in existing End2End tests - `Sep24End2EndTests`